homeassistant/components/broadlink/remote.py

```diff
@@ -233,27 +172,17 @@ async def async_send_command(self, command, **kwargs):
             raise ValueError(err_msg)
 
         at_least_one_sent = False
-        for _, codes in product(range(repeat), code_list):
-            if at_least_one_sent:
-                await asyncio.sleep(delay)
+        for _ in range(repeat):
+            for code in device.store.toggled_codes(code_list, subdevice):
```

Comment:
This logic changes the flag when the request doesn't work. We can only change the flag when the request is sent successfully or the device will not respond to the next command when it failed previously.

Reply:
No it doesnt change logic. The code is yielded. It is only toggled if the iteration continues.

Reply:
It doesn't work. The storage doesn't know about the request. If it failed and we ask for another code, it will change the flag. Another problem is that if we only request one code, it will not change the flag

Reply:
It does in a sense know, if the async_request fail with an exception, we break out of the loop. This will make the generate throw a GeneratorExit exception at the yield call, right before the flag was toggled, which mean it wont execute that toggle.

For the one code case, it will still execute the flag toggle. The code will continue after the yield to check if there is any more codes to yield, which there isnt.

There could be a bug obviously, but this should work identical to the old code.

Reply:
We could add a toggle_flag() method to the storage controller, and call it later when the operation is successful.

Reply:
Right, we should raise an exception instead of breaking. I dont want to redesign this code too much as part of this pull.

Reply:
I think it's better to toggle the flag explicitly instead of trying to guess inside the generator.

Reply:
since i needed in in two places i opted for abstracting it away. might have been a bad idea :). but should be okey now i think.